commit 2bbe0f25c6b905044803649a29df5f765f940b91
Author: Frank Denis <github@pureftpd.org>
Date:   Fri Oct 25 10:20:24 2024 +0200

    Fix code similar to the one fixed in 350d66fbbd6
    
    Noticed by @StayPirate, thanks!

diff --git a/src/ls.c b/src/ls.c
index 9063646..94150ad 100644
--- a/src/ls.c
+++ b/src/ls.c
@@ -972,8 +972,8 @@ void donlst(const char *base)
 
     if (*base != 0 && chdir(base) != 0) {
         if (*base++ == '-') {
-            while (!isspace((unsigned char) *base++));
-            while (isspace((unsigned char) *base++));
+            while (*base != 0 && !isspace((unsigned char) *base++));
+            while (*base != 0 && isspace((unsigned char) *base++));
             if (*base != 0 && chdir(base) != 0) {
                 addreply_noformat(550, MSG_STAT_FAILURE2);
                 return;
